topology/probes/opencontrail/routing_table.go

```diff
@@ -0,0 +1,339 @@
+				logging.GetLogger().Debugf("Delete interface %s from VRF %d", interfaceUUID, k)
+				vrf.InterfacesUUID[idx] = vrf.InterfacesUUID[len(vrf.InterfacesUUID)-1]
+				vrf.InterfacesUUID = vrf.InterfacesUUID[:len(vrf.InterfacesUUID)-1]
+				found = true
+				break
+			}
+		}
+		if found {
+			if len(vrf.InterfacesUUID) == 0 {
+				logging.GetLogger().Debugf("Delete VRF %d", k)
+				delete(mapper.routingTables, k)
+			}
+		}
+	}
+	return 0, errors.New("No VrfId was found")
+}
+
+func (mapper *OpenContrailProbe) OnInterfaceDeleted(interfaceUUID string) {
+	mapper.routingTableUpdaterChan <- RoutingTableUpdate{action: DelInterface, intf: interfaceUpdate{InterfaceUUID: interfaceUUID}}
+}
+
+// onRouteChanged writes the Contrail routing table into the
+// Contrail.RoutingTable metadata attribute.
+func (mapper *OpenContrailProbe) onRouteChanged(vrfId int) {
+	mapper.graph.Lock()
+	defer mapper.graph.Unlock()
+
+	vrf := mapper.getOrCreateRoutingTable(vrfId)
+
+	filter := graph.NewGraphElementFilter(filters.NewTermInt64Filter("Contrail.VRFID", int64(vrfId)))
+	intfs := mapper.graph.GetNodes(filter)
+
+	if len(intfs) == 0 {
+		logging.GetLogger().Debugf("No interface with VRF index %d was found (on route add)", vrfId)
+		return
+	}
+	for _, n := range intfs {
+		mapper.graph.AddMetadata(n, "Contrail.RoutingTable", vrf.Routes)
+		logging.GetLogger().Debugf("Update routes on node %s", n.ID)
+	}
+}
+
+func (mapper *OpenContrailProbe) addRoute(vrfId int, route OpenContrailRoute) {
+	vrf := mapper.getOrCreateRoutingTable(vrfId)
+	logging.GetLogger().Debugf("Adding route %v to vrf %d", route, vrfId)
+	for _, r := range vrf.Routes {
+		if r == route {
+			return
+		}
+	}
+	vrf.Routes = append(vrf.Routes, route)
+}
+
+func (mapper *OpenContrailProbe) delRoute(vrfId int, route OpenContrailRoute) {
+	vrf := mapper.getOrCreateRoutingTable(vrfId)
+	for i, r := range vrf.Routes {
+		if r.Prefix == route.Prefix {
+			logging.GetLogger().Debugf("Removing route %s from vrf %d ", r.Prefix, vrfId)
+			vrf.Routes[i] = vrf.Routes[len(vrf.Routes)-1]
+			vrf.Routes = vrf.Routes[:len(vrf.Routes)-1]
+			return
+		}
+	}
+	logging.GetLogger().Errorf("Can not remove route %v from vrf %d because route has not been found", route, vrfId)
+}
+
+// vrfInit uses the Contrail binary rt --dump to get all routes of a VRF.
+func (mapper *OpenContrailProbe) vrfInit(vrfId int) error {
+	logging.GetLogger().Debugf("Initialisation of VRF %d...", vrfId)
+
+	cmd := exec.Command("rt", "--dump", fmt.Sprint(vrfId))
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return err
+	}
+	cmd.Start()
+	defer cmd.Wait()
+
+	scanner := bufio.NewScanner(stdout)
+	separator := regexp.MustCompile("[[:space:]]+")
+
+	// Remove the rt --dump stdout header
+	scanner.Scan()
+	scanner.Scan()
+	scanner.Scan()
+
+	for scanner.Scan() {
+		s := separator.Split(scanner.Text(), -1)
+		// Ignore non complete entries
+		if len(s) != 6 {
+			continue
+		}
+
+		destination := strings.Split(s[0], "/")
```

Comment:
need to check that the destination split give the right number of entries ?

Reply:
This split is useless in fact. So i just remove it.